app/validators/questions/question_validator.py

```diff
@@ -32,3 +36,14 @@ def _validate_answer_labels(self):
                 self.add_error(
                     self.ANSWER_LABEL_MISSING_MULTIPLE_ANSWERS, answer_id=answer["id"]
                 )
+
+    def _validate_q_codes(self):
+        for answer in self.answers:
+            if answer.get("options") and answer.get("type") != "Radio":
```

Comment:
This should checkbox explicitly, as Dropdown can have options.

Reply:
Added explicit rule for Checkbox, also dropdown is being checked properly now.